### PR TITLE
Export raw validator function

### PR DESCRIPTION
**Pull request by tomohisaota**
Raw validator function can be used for type guard

I want to use multiple validators for type guard
example:
if(rawValidatorForType1(data)){
 // do something for Type1
}
else if(rawValidatorForType2(data)){
 // do something for Type2
}

And I prefer to use raw validator which does not throw exception for invalid input.

### CODE REVIEW

File: src/template.ts [hunk 1]
```diff
@@ -62,7 +62,10 @@ export const compileSchema = (schemaName: string, typeName: string) =>
 export const validateFn = (
   typeName: string,
   schemaName: string,
-) => `const rawValidate${typeName} = ${compileSchema(schemaName, typeName)};
+) => `export const rawValidate${typeName} = ${compileSchema(
```
Comment (ForbesLindesay) [inline src/template.ts]: I wonder about renaming this to `is{typeName}` before we export it and lock in the function name as part of the public API.

**Comment by tomohisaota** [comment]
That will be even better:-)

Actually, I have one more idea.
Can we add optional boolean parameter "throwsOnError" to the is${typeName} method?

My usecase
if(isTypeA(data)){
// do something for typeA
}
else if(isTypeB(data)){
// do something for typeB
}
else if(isTypeC(data,true)){
// do something for typeC
// There're no other cases since it throws error on validation failure
}

**Comment by tomohisaota** [comment]
I think I can just add another function 

```
export function isTypeA(data: unknown, throwsOnError:boolean = false): data is TypeA {
  if (rawValidateTypeA(data)) {
    return true;
  } else {
    if(!throwsOnError){
      return false
    }
    throw new Error(
        ajv.errorsText(rawValidateTypeA.errors, {
          dataVar: "NTEResponseLattice"
        }) +
        "\n\n" +
        inspect(value)
    );
  }
}

```

I will try to update the pull req when I have spare time. (within 24h)

**Comment by tomohisaota** [comment]
I had to increase jest test timeout because build-parameters.test.ts fails all the time.

**Comment by tomohisaota** [comment]
I have cleaned up commits.

I have increased jest test timeout again to pass unit tests.
I believe my change does not affect test time though...

**Comment by tomohisaota** [comment]
Any feedback?

The pull req is marked as Changes requested, but the request is not valid anymore.
Do I need to create another clean pull req?

**Comment by tomohisaota** [comment]
I can do nothing about the change request.
Do you want me to recreate the pull req?

**Comment by ForbesLindesay** [comment]
Thanks for your patience with this pull request. I think it's a really valuable addition, I've just been struggling to keep up with all the open source stuff lately. This should be automatically released by CI shortly.

**Comment by tomohisaota** [comment]
Glad that I can help, and thank you for creating such a useful library :-)
TypeScript is not good at handling unknown input, and the library filled the missing piece!!

**Comment by ForbesLindesay** [comment]
Thanks. I've added you as a collaborator to this repo in case you're interested in helping out. Please don't merge your own PRs until approved, but feel free to review/approve/merge other people's PRs. In order to trigger a release, the title of the commit title when you squash and merge must start with `feat: ` or `fix: ` and to trigger a major version, you use `BREAKING CHANGE: <explanation of the breaking change>` in the body of the commit message.

**Pull request merged**